### PR TITLE
convert to Dist::Zilla

### PR DESCRIPTION
**Pull request by skaji**
# Description

Converting to Dist::Zilla. I hope this makes perl-crypt-openssl-x509 more maintainable.

* I tried following your dist.ini in your other distributions, such as https://github.com/jonasbn/perl-workflow
* I used [MakeMaker::Awesome](https://metacpan.org/pod/Dist::Zilla::Plugin::MakeMaker::Awesome) plugin to generate Makefile.PL.
  * We should not change Makefile.PL manually; instead, change maint/Makefile_header.PL so that it will be injected to Makefile.PL by `dizl build`.
  * I added code of hints/MSWin32.pl in maint/Makefile_header.PL, and removed hints/MSWin32.pl.
* I think I prepared a "basic" dist.ini file; so please feel free to change dist.ini as you like.

## Type of change

Authoring tool change

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Changes have been manually tested (please provide information on test platform using the fields below)

## Test / Development Platform Information

- Operating system and version: macOS 12.1
- Crypt::OpenSSL::X509 version: 1.912
- Perl version: 5.34.0
- OpenSSL version: 1.1.1m / 3.0.1

### CODE REVIEW

File: dist.ini [hunk 1]
```diff
@@ -0,0 +1,60 @@
+; REF: Dist::Zilla https://metacpan.org/pod/Dist::Zilla
+name = Crypt-OpenSSL-X509
+author = Dan Sully
```
Comment (timlegge) [inline dist.ini:3]: you can add additional author lines maybe add Jonas if he thinks its warranted.

I have used the following in some of my projects where it made sense

[Meta::Maintainers]
maintainer = 

[Meta::Contributors]
contributor =

File: dist.ini [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+[Test::CPAN::Meta::JSON]
+
+; REF: https://metacpan.org/pod/Dist::Zilla::Plugin::Prereqs::FromCPANfile
+[Prereqs::FromCPANfile]
+
+; REF: https://metacpan.org/pod/Dist::Zilla::Plugin::CopyFilesFromBuild
+[CopyFilesFromBuild]
+copy = Makefile.PL
+copy = LICENSE
+
```
Comment (timlegge) [inline dist.ini:56]: I like some of the git plugins like:

[Git::GatherDir]
exclude_filename = cpanfile
exclude_filename = Makefile.PL
exclude_filename = dev-bin/cpanm
exclude_filename = Dockerfile
exclude_filename = MANIFEST
exclude_filename = README

[Git::Tag]
tag_format  = %V       ; this is the default
tag_message = %V       ; this is the default

[Git::Commit]
changelog = Changes      ; this is the default

I also add gpg signatures to all my modules now:

[Signature]

File: maint/Makefile_header.PL [hunk 3]
```diff
@@ -0,0 +1,43 @@
+use Config;
```
Comment (timlegge) [inline maint/Makefile_header.PL:1]: Interesting.  I may steal this for one of my modules that I was having issues moving to Dist::Zilla

**Review by timlegge — COMMENTED** [review]
@skaji I like this a lot.  I support a module or two with a very similar build.  I may "borrow" this for there too...

File: dist.ini (hunk 1 above)
Comment (jonasbn) [reply dist.ini:3]:
```
[Meta::Maintainers]
maintainer =

[Meta::Contributors]
contributor =
```

Looks good, I will populate that accordingly.

**Review by jonasbn — APPROVED** [review]
LGTM

I will make some minor adjustments and fill in some blanks, thanks for taking the time to check my Dist::Zilla approach - it might not be perfect, but it makes adoption easier for me. I am of course open to improvements.

**Comment by skaji** [comment]
@timlegge @jonasbn Thank you both!

**Comment by timlegge** [comment]
@jonasbn I have been testing against macos and have this issue with a similar Dist::Zill config https://github.com/timlegge/perl-Crypt-OpenSSL-VerifyX509/runs/5265302826?check_suite_focus=true

Looking at resolving now

**Comment by jonasbn** [comment]
@timlegge please let me know how it works out. I would love to _chat_ at some point and pick your brains on several issues in regard to the XS issues I observe.

**Comment by timlegge** [comment]
@jonasbn actually it appears to be an issue installing the current cpan version of Crypt::OpenSSL::X509 with 

runs-on macos-latest using:

https://github.com/timlegge/perl-Crypt-OpenSSL-VerifyX509/blob/dist-zilla/.github/workflows/macos.yml

I used your version as for https://github.com/timlegge/perl-Crypt-OpenSSL-VerifyX509/blob/dist-zilla/.github/workflows/macos-10.15.yml and it at least compiles now.  So it is not related to this PR.

I am available most of the time on as timlegge on irc - libera-chat, OFTC and perl email and almost anything else works too.

**Pull request merged**